lesson_05/lesson_05.cpp

```diff
@@ -0,0 +1,273 @@
+	february,
+	march,
+	april,
+	may,
+	june,
+	july,
+	august,
+	september,
+	october,
+	november,
+	december
+};
+
+int main()
+{
+	std::cout << "Home Work 5\n";
+	std::cout << "-----------\n\n";
+
+	int user_choice{ 0 };
+	while (user_choice != 7)
+	{
+		print_menu();
+		std::cin >> user_choice;
+		clear_console();
+		switch (user_choice)
+		{
+		case 1:
+			max_from_three_numbers();
+			break;
+		case 2:
+			min_from_two_numbers();
+			break;
+		case 3:
+			division_to_5_and_11();
+			break;
+		case 4:
+			triangle_from_angles();
+			break;
+		case 5:
+			month_num_to_name();
+			break;
+		case 6:
+			month_num_to_season();
+			break;
+		case 7:
+			std::cout << "Exit\n";
+			break;
+		default:
+			std::cout << "Wrong choice!";
+			user_choice = 0;
+			break;
+		}
+		std::cout << "\nPress Enter to continue" << std::flush;
+		std::cin.clear();
+		std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+		std::cin.get();
+		clear_console();
+	}
+
+	return 0;
+}
+
+
+void print_menu()
+{
+	std::cout << "1. max from three numbers\n";
+	std::cout << "2. min from two numbers\n";
+	std::cout << "3. division to 5 and 11\n";
+	std::cout << "4. triangle from angles\n";
+	std::cout << "5. month num to name\n";
+	std::cout << "6. month num to season\n";
+	std::cout << "7. Exit\n\n";
+
+	std::cout << "Enter number:";
+}
+
+void clear_console() {
+#ifdef _WIN32
+	std::system("cls");
+#else
+	std::cout << "\033[2J\033[1;1H";
+#endif
+}
+
+void max_from_three_numbers()
+{
+	double a{ 0 }, b{ 0 }, c{ 0 }, max{ std::numeric_limits< double >::lowest() };
+
+	std::cout << "Enter number 1:\n";
+	std::cin >> a;
+	if (a > max)
+	{
+		max = a;
+	}
+
+	std::cout << "Enter number 2:\n";
+	std::cin >> b;
+	if (b > max)
+	{
+		max = b;
+	}
+
+	std::cout << "Enter number 3:\n";
+	std::cin >> c;
+	if (c > max)
+	{
+		max = c;
+	}
+
+	std::cout << std::endl;
+
+	std::cout << "max from [" <<a <<", " << b << ", " << c <<"] is " << max << std::endl;
+}
+
+void min_from_two_numbers()
+{
+	double a{ 0 }, b{ 0 };
+
+	std::cout << "Enter number 1:\n";
+	std::cin >> a;
+
+	std::cout << "Enter number 2:\n";
+	std::cin >> b;
+
+	std::cout << std::endl;
+
+	std::cout << "min from [" << a << ", " << b << "] is " << (a < b ? a : b) << std::endl;
+}
+
+void division_to_5_and_11()
+{
+	int number{ 0 };
+
+	std::cout << "Enter number:\n";
+	std::cin >> number;
+
+	if (number % 5 == 0 && number % 11 == 0) {
+		std::cout << "number " << number <<" is div to 5 and 11" << std::endl;
+	}
+	else {
+		std::cout << "number " << number << " is not div to 5 and 11" << std::endl;
+	}
+}
+
+void triangle_from_angles()
+{
+	double angle1{ 0 }, angle2{ 0 }, angle3{ 0 };
+
+	std::cout << "Enter angle 1:\n";
+	std::cin >> angle1;
+
+	std::cout << "Enter angle 2:\n";
+	std::cin >> angle2;
+
+	std::cout << "Enter angle 3:\n";
+	std::cin >> angle3;
+
+	double diff{ 180 - angle1 - angle2 - angle3 };
+	diff = diff < 0 ? -diff : diff;
+
+	if (diff <= std::numeric_limits< double >::epsilon()) {
```

Comment:
checking with <= sum of numbers will no require epsilon check.
angles 180, 0, 0 are not valid but will pass your check

Reply:
fixed